clarity/contracts/arkadiko-liquidation-rewards-diko-v1-1.clar

```diff
@@ -0,0 +1,106 @@
+(use-trait liquidation-rewards-trait .arkadiko-liquidation-rewards-trait-v1.liquidation-rewards-trait)
+
+;; Errors
+(define-constant ERR-NOT-AUTHORIZED u34401)
+(define-constant ERR-EPOCH-NOT-ENDED u34002)
+(define-constant ERR-EMERGENCY-SHUTDOWN-ACTIVATED u34003)
+
+;; Variables
+;; TODO - SET FOR MAINNET
+(define-data-var end-epoch-block uint (+ block-height u720)) ;; 5 days after deploy
+(define-data-var epoch-rate uint u100000) ;; 10%
+(define-data-var blocks-per-epoch uint u720) ;; 5 days
+(define-data-var shutdown-activated bool false)
+
+(define-read-only (get-end-epoch-block)
+  (ok (var-get end-epoch-block))
+)
+
+(define-read-only (get-epoch-rate)
+  (ok (var-get epoch-rate))
+)
+
+(define-read-only (get-blocks-per-epoch)
+  (ok (var-get blocks-per-epoch))
+)
+
+(define-read-only (get-epoch-info)
+  (ok {
+    blocks: (var-get blocks-per-epoch),
+    rate: (var-get epoch-rate),
+    end-block: (var-get end-epoch-block)
+  })
+)
+
+(define-read-only (get-shutdown-activated)
+  (or
+    (unwrap-panic (contract-call? .arkadiko-dao get-emergency-shutdown-activated))
+    (var-get shutdown-activated)
+  )
+)
+
+;; @desc toggles the killswitch
+(define-public (toggle-shutdown)
+  (begin
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-guardian-address)) (err ERR-NOT-AUTHORIZED))
+
+    (ok (var-set shutdown-activated (not (var-get shutdown-activated))))
+  )
+)
+
+;; ---------------------------------------------------------
+;; Add rewards
+;; ---------------------------------------------------------
+
+;; @desc Amount of rewards that can be added
+(define-read-only (get-rewards-to-add)
+  (let (
+    (total-block-rewards (contract-call? .arkadiko-diko-guardian-v1-1 get-staking-rewards-per-block))
+
+    (rewards-to-add (/ (* total-block-rewards (var-get blocks-per-epoch)) (var-get epoch-rate)))
+  )
+    (if (< block-height (var-get end-epoch-block))
+      u0
+      rewards-to-add
+    )
+  )
+)
+
+;; @desc add DIKO rewards to liquidation-rewards
+(define-public (add-rewards (liquidation-rewards <liquidation-rewards-trait>))
```

Comment:
Q: we/someone has to call this once per epoch?

Reply:
Correct. Can be done retroactively if rewards were not added during epoch.